R/google_sheets.R

```diff
@@ -18,39 +21,66 @@ uploadGoogleSheet <- function(filepath, title, overwrite = FALSE){
 #' @param treatTheseAsNA - character vector that each item represents NA
 #' @param firstRowAsHeader - argument to control if you want to treat first row as header
 #' @param commentChar - treat the character as comment.
+#' @param tokenFileId - No longer used. It was kept for backward compatibility for the old Desktop Versions that don't handle OAuth token in server side.
 #' @param guessDataType - flag to tell if you want googlesheets::gs_read to guess column data type
+#' @param tzone - timezone
+#' @param id - ID off the sheet
 #' @export
-getGoogleSheet <- function(title, sheetName, skipNRows = 0, treatTheseAsNA, firstRowAsHeader = TRUE, commentChar, tokenFileId=NULL, guessDataType=TRUE, ...){
-  if(!requireNamespace("googlesheets")){stop("package googlesheets must be installed.")}
+getGoogleSheet <- function(title, sheetName, skipNRows = 0, treatTheseAsNA = NULL, firstRowAsHeader = TRUE, commentChar, tokenFileId=NULL, guessDataType=TRUE, tzone=NULL, id=NULL, ...){
+  if(!requireNamespace("googlesheets4")){stop("package googlesheets4 must be installed.")}
+  if(!requireNamespace("googledrive")){stop("package googledrive must be installed.")}
   if(!requireNamespace("stringr")){stop("package stringr must be installed.")}
 
   token <- getGoogleTokenForSheet(tokenFileId)
-  googlesheets::gs_auth(token)
-  gsheet <- googlesheets::gs_title(title)
+  googlesheets4::sheets_set_token(token)
+  googledrive::drive_set_token(token)
+  if(!is.null(id)) {
```

Comment:
Please add a comment on why we do this switch.

Reply:
Added comment.